libraries/BLE/src/BLEClient.cpp

```diff
@@ -142,9 +153,10 @@ bool BLEClient::connect(BLEAddress address, esp_ble_addr_type_t type) {
     return false;
   }
 
-  rc = m_semaphoreOpenEvt.wait("connect");   // Wait for the connection to complete.
-  // check the status of the connection and cleanup in case of failure
-  if (rc != ESP_GATT_OK) {
+  bool got_sem = m_semaphoreOpenEvt.timedWait("connect", timeoutMs);   // Wait for the connection to complete.
+	rc = m_semaphoreOpenEvt.value();
+	// check the status of the connection and cleanup in case of failure
+	if (!got_sem || rc != ESP_GATT_OK) {
```

Comment:
```suggestion
  bool got_sem = m_semaphoreOpenEvt.timedWait("connect", timeoutMs);   // Wait for the connection to complete.
  rc = m_semaphoreOpenEvt.value();
  // check the status of the connection and cleanup in case of failure
  if (!got_sem || rc != ESP_GATT_OK) {
```